global_planner/src/planner_core.cpp

```diff
@@ -308,6 +308,47 @@ uint32_t GlobalPlanner::makePlan(const geometry_msgs::PoseStamped& start, const
     bool found_legal = planner_->calculatePotentials(costmap_->getCharMap(), start_x, start_y, goal_x, goal_y,
                                                     nx * ny * 2, potential_array_);
 
+
+    geometry_msgs::PoseStamped best_pose = goal;
+    if (!found_legal) {
+      double resolution = costmap_->getResolution();
+      geometry_msgs::PoseStamped p = goal;
+
+      double best_sdist = DBL_MAX;
+
+      unsigned int mx, my;
+      for(double dy = -tolerance; dy <= tolerance; dy += resolution){
```

Comment:
S: we could change to go from distance zero to outwards (distance tolerance) and break if it is legal. this way we don't compute for all points in the circle

Reply:
we're searching a circle so to terminate early, we'd have to iterate from inside out, i.e. on concentric circles of cells, with increasing radius? I think the added complexity would not be justified (unless profiling revealed that this is a bottleneck somehow / or if you know an elegant implementation?)

Reply:
yes, inside out. 
What i thought is to check adjacent cells, since it is a grid. A function to return the adjacent cells and store the already visited ones. 
The number of loops depends on the resolution. If resolution is 1 and radius is 5, you have 5 iterations (5 adjacent)

![image](https://user-images.githubusercontent.com/6097267/215423120-81f6fdce-d279-4965-bed1-a64ededda98f.png)

Reply:
Well I can see at least 2 problems:
- Above picture seems to suggest that after finding a cell on the purple rectangle, we can stop the iteration. This is not true though because the corner is `sqrt(3^2+3^2) = sqrt(18)` away, which is further than half of the orange cells (4 cells have distance `4`, and 8 cells have distance `sqrt(17)`
- even if we changed the iteration to go through the cells in ascending distance order (i.e. first the 4 direct neighbors 1 cell away, then the ones sqrt(2) away, then 2, then sqrt(5), etc), this still wouldn't work because the goal is generally not even in the center of the green cell?